app/models/discount_code.py

```diff
@@ -13,7 +13,9 @@
 class DiscountCode(SoftDeletionModel):
     __tablename__ = "discount_codes"
     __table_args__ = (
-        UniqueConstraint('event_id', 'code', name='uq_event_discount_code'),
+        UniqueConstraint(
+            'event_id', 'code', 'deleted_at', name='uq_event_discount_code_deleted_at'
```

Comment:
Don't add deleted_at in constraint name, it's an implementation detail

Reply:
Remove from all?

Reply:
Yes